src/util/imagefiledata.cpp

```diff
@@ -0,0 +1,71 @@
+#include "util/imagefiledata.h"
+
+#include <QBuffer>
+#include <QByteArray>
+#include <QFile>
+#include <QImage>
+#include <QImageReader>
+
+#include "util/assert.h"
+
+ImageFileData::ImageFileData(
+        const QByteArray& coverArtBytes, const QByteArray& coverArtFormat)
+        : QImage(QImage::fromData(coverArtBytes)),
+          m_coverArtBytes(coverArtBytes) {
+    m_coverArtFormat = coverArtFormat.isNull() ? readFormatFrom(m_coverArtBytes) : coverArtFormat;
```

Comment:
```suggestion
    m_coverArtFormat = coverArtFormat.isEmty() ? readFormatFrom(m_coverArtBytes) : coverArtFormat;
```
to match the assertion in saveFileAddSuffix()